pilot/pkg/proxy/envoy/v2/ads_common.go

```diff
@@ -19,6 +19,32 @@ import (
 	"istio.io/istio/pkg/config/schema/collections"
 )
 
+// PushAffectsProxy checks if a pushEv will affect a specified proxy. That means whether the push will be performed
+// towards the proxy.
+func PushAffectsProxy(pushEv *XdsEvent, proxy *model.Proxy) bool {
+	if proxy.Type != model.SidecarProxy {
+		return true
+	}
+	if len(pushEv.configsUpdated) == 0 {
+		return true
+	}
+
+	for kind, resources := range pushEv.configsUpdated {
+		if len(resources) == 0 {
+			return true
+		}
+
+		for name := range resources {
+			ok, scoped := proxy.SidecarScope.DependsOnConfig(kind, name)
```

Comment:
Because currently `SidecarScope` only tells us three types of configs(service, virtual service, destination rule) dependencies. And whether other types of configs affect the proxies in this scope is unknown.
Thus I involve a `scoped` return value to indicate that. Maybe later our `SidecarScope` will do more and hold more dependency info of other types.

Reply:
For unknown dependencies, we can just return true. We do not actually care about that, but donot want to miss one push.

Reply:
I just want the `SidecarScope` to take a single responsibility and tell what it knows. How to handle unknown/unscoped case is leaved to the caller.